frontend/public/components/catalog.jsx

```diff
@@ -0,0 +1,214 @@
+      const tileName = serviceClassDisplayName(serviceClass);
+      const tileIconClass = getCSCIcon(serviceClass);
+      const tileImgUrl = getCSCImage(serviceClass, tileIconClass);
+      const tileDescription = _.get(serviceClass, 'spec.description');
+      const tileProvider = _.get(serviceClass, 'spec.externalMetadata.providerDisplayName');
+
+      return {
+        resource: serviceClass,
+        kind,
+        tileName,
+        tileIconClass,
+        tileImgUrl,
+        tileDescription,
+        tileProvider
+      };
+    });
+  }
+
+  normalizeImagestreams(imagestreams, kind) {
+    imagestreams = _.map(imagestreams, imagestream => {
+      // A valid specTag has a 'builder' tag. no 'hidden' tag, and exists in 'status.tags'.
+      const builderSpecTagName = getBuilderSpecTagName(imagestream);
+      if (builderSpecTagName) {
+        const tileName = _.get(imagestream, ['metadata', 'annotations', 'displayName']) || imagestream.metadata.name;
+        // TODO: icon, description and provider should be based on selected version?
+        const tileIconClass = getImageStreamTagAnnotation(imagestream, 'iconClass', builderSpecTagName);
+        const tileDescription = getImageStreamTagAnnotation(imagestream, 'description', builderSpecTagName);
+        const tileProvider = getImageStreamTagAnnotation(imagestream, 'openshift.io/provider-display-name', builderSpecTagName);
+        return {
+          resource: imagestream,
+          kind,
+          tileName,
+          tileIconClass,
+          tileDescription,
+          tileProvider
+        };
+      }
+      return {
+        hidden: true
+      };
+    });
+
+    return _.filter(imagestreams, imagestream => {
+      return !imagestream.hidden;
+    });
+  }
+
+  handleFilterChange(event) {
+    this.setState({filterValue: event.target.value});
+  }
+
+  clearFilter() {
+    this.setState({filterValue: ''});
+  }
+
+  render() {
+    const {loaded, loadError} = this.props;
+    const {filteredItems} = this.state;
+
+    return <div className="co-m-pane">
+      <div className="co-m-pane__body">
+        <StatusBox data={filteredItems} loaded={loaded} loadError={loadError} label="Resources">
+          <CatalogList items={filteredItems} />
+        </StatusBox>
+      </div>
+    </div>;
+  }
+}
+
+CatalogListPage.displayName = 'CatalogList';
+
+CatalogListPage.propTypes = {
+  obj: PropTypes.object
+};
+
+// eventually may use namespace
+// eslint-disable-next-line no-unused-vars
+export const Catalog = ({namespace, flags}) => {
+  const resources = [];
+  flags.SERVICE_CATALOG
+    ? resources.push({
+      isList: true,
+      kind: 'ClusterServiceClass',
+      namespaced: false,
+      prop: 'clusterserviceclasses'
+    })
+    : resources;
+  flags.OPENSHIFT
+    ? resources.push({
+      isList: true,
+      kind: 'ImageStream',
+      namespace: 'openshift',
+      prop: 'imagestreams'
+    })
+    : resources;
+  return <div className="catalog">
```

Comment:
Why the ternary's when the else doesn't do anything?

Reply:
oops..initially I was assigning it to a var